entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -2,17 +2,545 @@
+20200910201708_Two
+20200910202050_Three (Pending)
+ajcvickers@avickers420u:~/AllTogetherNow/Daily$
+```
+
+In addition, there is now `Get-Migations` command for the Package Manager Console with the same functionality.
+
+### ModelBuilder API for value comparers
+
+EF Core properties for custom mutable types [require a value comparer](xref:core/modeling/value-comparers) for property changes to be detected correctly. This can now be specified as part of configuring the value conversion for the type. For example:
+
+```c#
+modelBuilder
+    .Entity<EntityType>()
+    .Property(e => e.MyProperty)
+    .HasConversion(
+        v => JsonSerializer.Serialize(v, null),
+        v => JsonSerializer.Deserialize<List<int>>(v, null),
+        new ValueComparer<List<int>>(
+            (c1, c2) => c1.SequenceEqual(c2),
+            c => c.Aggregate(0, (a, v) => HashCode.Combine(a, v.GetHashCode())),
+            c => c.ToList()));
+```
+
+### EntityEntry TryGetValue methods
+
+A `TryGetValue` method has been added to `EntityEntry.CurrentValues` and `EntityEntry.OriginalValues`. This allows the value of a property to be requested without first checking if the property is mapped in the EF model. For example:
+
+```c#
+if (entry.CurrentValues.TryGetValue<string>(propertyName, out var value))
+{
+    Console.WriteLine(value);
+}
+```
+
+This feature was contributed by [@m4ss1m0g](https://github.com/m4ss1m0g). Many thanks!
+
+### Default max batch size for SQL Server
+
+The default maximum batch size for SaveChanges on SQL Server is now 42. This value was obtained through [analysis of batching performance](https://github.com/dotnet/efcore/issues/9270), and not because we have discovered the Ultimate Question to Live, the Universe, and Everything.
+
+### Default environment to Development
+
+The EF Core command line tools now automatically configure the `DOTNET_ENVIRONMENT` and `ASPNETCORE_ENVIRONMENT` environment variables to "Development". This improves the experience for working with ASP.NET Core during development. See [#19903](https://github.com/dotnet/efcore/issues/19903).
+
+### Better migrations column ordering
+
+The columns for un-mapped base classes are now ordered after other columns for mapped entity types. Note this only impacts newly created tables. The column order for existing tables remains unchanged. See [#11314](https://github.com/dotnet/efcore/issues/11314).
+
+### Query improvements
+
+EF Core 5.0 RC1 contains some additional query translation improvements:
+
+* Translation of `is` on Cosmos--see [#16391](https://github.com/dotnet/efcore/issues/16391)
+* User mapped functions can now be annotated to control null propagation--see [#19609](https://github.com/dotnet/efcore/issues/19609)
+* Support for translation of GroupBy with conditional aggregates--see [#11711](https://github.com/dotnet/efcore/issues/11711)
+* Translation of Distinct operator over group element before aggregate--see [#17376](https://github.com/dotnet/efcore/issues/17376)
```

Comment:
```suggestion
* Translation of Distinct over group elements before aggregate--see [#17376](https://github.com/dotnet/efcore/issues/17376)
```

Reply:
But Distinct is applied over the group element 'g'.
There are inner elements in the group element.

Reply:
Leaving as is for now.

Reply:
OK, just that English-wise I don't think it works as-is, but not very important.

Reply:
There is nothing wrong English-wise here. You can apply Distinct over any enumerable stream.